Create an istio registry per primary

nrfox opened the pull request:
This change does a few main things:

1. Polls the istio registry for the registry services instead of fetching these lazily. Similar to how proxy status was implemented.
2. Polls every primary across clusters for proxy status. Polls a single primary in each cluster for registry services. Registry services are shared across revisions.
3. Adds a "multi-primary" testing workflow and a couple e2e tests for multi-primary.

Fixes #6432 
Fixes #5692

jshaughn commented on store/store.go:
```diff
@@ -0,0 +1,32 @@
+// Store is a generic key value store. Implementations should be safe for concurrent use.
+// It is basically just a map with a lock. Does not yet implement Set so we don't have to
+// worry about this thing growing without bound.
+type Store[T any] interface {
```
OK, so I learned that Golang Maps are not threadsafe :)

nrfox replied:
It's thread safe for readers but not if you have something concurrently writing to the map while other things are trying to read from it.

nrfox commented on business/istio_validations.go:
```diff
@@ -106,8 +102,8 @@ func (in *IstioValidationsService) GetValidations(ctx context.Context, cluster,
 
 	if istioApiEnabled {
-		// @TODO registry services for remote cluster
-		go in.fetchRegistryServices(&registryServices, errChan, &wg)
+		criteria := RegistryCriteria{AllNamespaces: true, Cluster: cluster}
+		registryServices = in.businessLayer.RegistryStatus.GetRegistryServices(criteria)
```
Perhaps not. I'll try and remove this and test with istio api enabled = false.

nrfox commented on business/layer.go:
```diff
@@ -41,43 +39,15 @@ var (
 	tracingClient    tracing.ClientInterface
 	kialiCache       cache.KialiCache
 	prometheusClient prometheus.ClientInterface
+	poller           ControlPlaneMonitorer
```
I just stuttered a bit whilst typing

nrfox commented:
@jshaughn addressed feedback and tested with istio api disabled. Note that I renamed the github multicluster workflow to be multicluster primary-remote so if this gets merged then the "required checks" will need to get updated.

jshaughn reviewed (commented):
Everything looks good to me, my only question is still around the polling interval.  It looks like we are piggybacking on `conf.KubernetesConfig.CacheTokenNamespaceDuration`.  Is that appropriate or should we have a dedicated polingInterval for this?  The default for `CacheTokenNamespaceDuration` looks like 10s, is that maybe too frequent?

nrfox commented:
> Everything looks good to me, my only question is still around the polling interval. It looks like we are piggybacking on `conf.KubernetesConfig.CacheTokenNamespaceDuration`. Is that appropriate or should we have a dedicated polingInterval for this? The default for `CacheTokenNamespaceDuration` looks like 10s, is that maybe too frequent?

Before this piggybacked on `CacheDuration`: https://github.com/kiali/kiali/blob/96ea04c47e69e7429f3ee7e9431cb5599c7ed9bb/kubernetes/cache/cache.go#L95 which by default is `300s`: https://github.com/kiali/kiali/blob/96ea04c47e69e7429f3ee7e9431cb5599c7ed9bb/config/config.go#L777 imo this is too high. You're much more likely to notice stale data if it's 5 minute stale. I think 15-30s is reasonable. You probably won't notice stale data at that rate but it's not too frequent. A separate config value makes sense. I'll put it under `external_services.istio`.

jshaughn reviewed (approved):
I got nothing else!

matejnesuta reviewed (approved):
LGTM. We can always change the test execution for multi-primary and primary-remote later.
